packages/common-all/src/store/IDataQuery.ts

```diff
@@ -0,0 +1,39 @@
+import { ResultAsync } from "neverthrow";
+import {
+  DendronError,
+  NoteChangeEntry,
+  NotePropsByIdDict,
+  NotePropsMeta,
+  SchemaModuleDict,
+  SchemaModuleProps,
+} from "..";
+
+export type INoteQueryOpts = {
```

Comment:
I don't think we should expose `onlyDirectChildren` - it adds too much bespoke logic into the store/engine side of things and muddies the interface.  Instead, I think it's better to have the calling side apply these kinds of filtering. Same goes with the word `originalQS` - it should just be `queryString`

Reply:
it might make sense but see comment above - i'm trying to avoid large refactorings with this change. this is meant to be an incremental refactor to get us to the desired state of having a more fully abstracted store